src/v0.js

```diff
@@ -11,15 +11,36 @@ const watchpb = require('./authzedapi/authzed/api/v0/watch_service_pb.js')
 
 const util = require('./util')
 
-function Client(token, endpoint = util.authzedEndpoint, insecure = false) {
+function NewClient(token, endpoint = util.authzedEndpoint, insecure = false) {
   const creds = util.createClientCreds(token, insecure);
-  this.acl = new aclsvc.ACLServiceClient(endpoint, creds);
-  this.ns = new nssvc.NamespaceServiceClient(endpoint, creds);
-  this.watch = new watchsvc.WatchServiceClient(endpoint, creds);
+
+  const acl = new aclsvc.ACLServiceClient(endpoint, creds);
+  const ns = new nssvc.NamespaceServiceClient(endpoint, creds);
+  const watch = new watchsvc.WatchServiceClient(endpoint, creds);
+
+  const handler = {
+    get(target, name) {
+      if (acl[name]) {
+        return acl[name];
+      }
+
+      if (ns[name]) {
+        return ns[name];
+      }
+
+      if (watch[name]) {
+        return watch[name];
+      }
+
+      return undefined;
+    }
+  }
+
+  return new Proxy({}, handler);
```

Comment:
If we generate typescript headers, is this going to cause friction?

Reply:
No, since the headers will specify the proper types; how they are implemented is fairly immaterial